test/unit/test_iostream.c

```diff
@@ -0,0 +1,125 @@
+/*
+ * SPDX-License-Identifier: BSD-2-Clause
+ *
+ * Copyright 2010-2021, Tarantool AUTHORS, please see AUTHORS file.
+ */
+
+#include <sys/uio.h>
+#include <fcntl.h>
+#include <sys/errno.h>
+
+#include "test_iostream.h"
+#include "trivia/util.h"
+
+static ssize_t
+test_stream_read(struct iostream *io, void *buf, size_t count)
+{
+	struct test_stream_ctx *ctx = io->ctx;
+	io->fd = ctx->rfd;
+	(void)buf;
+	ssize_t nrd = 0;
+	if (ctx->avail_rd == 0)
+		return IOSTREAM_WANT_READ;
+	nrd = MIN(count, ctx->avail_rd);
+	ctx->avail_rd -= nrd;
+	return nrd;
+}
+
+static ssize_t
+test_stream_write(struct iostream *io, const void *buf, size_t count)
+{
+	assert(io->fd >= 0);
```

Comment:
Better use `fail` in tests so that checks are not compiled out in release builds.

Reply:
Indeed. Thanks for the tip!